packages/core/src/state-management/repository.ts

```diff
@@ -0,0 +1,85 @@
+import { Document } from '../document';
+import DocID from '@ceramicnetwork/docid';
+import { AsyncLruMap } from './async-lru-map';
+import { StateStore } from '../store/state-store';
+import { DocumentFactory } from './document-factory';
+import { DocState } from '@ceramicnetwork/common';
+
+export class Repository {
+  readonly #map: AsyncLruMap<Document>;
+  #documentFactory?: DocumentFactory;
+  #stateStore?: StateStore
+
+  constructor(limit: number) {
+    this.#map = new AsyncLruMap(limit, async (entry) => {
+      await entry.value.close();
+    });
+  }
+
+  // This should not be here, but here we are with the initialization process.
```

Comment:
```suggestion
  // Ideally this would be provided in the constructor, but circular dependencies in our initialization process make this necessary for now
```